src/clients/SpokePoolClient.ts

```diff
@@ -617,6 +670,20 @@ export class SpokePoolClient extends BaseAbstractClient {
     return depositEvent.args.quoteTimestamp > currentTime;
   }
 
+  // Temporary type discriminator for v2 -> v3 transition.
+  protected isV3DepositEvent(event: FundsDepositedEvent | V3FundsDepositedEvent): event is V3FundsDepositedEvent {
+    const { event: eventName } = event;
+    assert(isDefined(eventName) && ["FundsDeposited", "V3FundsDeposited"].includes(eventName));
```

Comment:
Why assert? Don't the FundsDepositedEvent and V3FundsDepositedEvent types already have the eventName field?

Reply:
Definition here: https://github.com/ethers-io/ethers.js/blob/f97b92bbb1bde22fcc44100af78d7f31602863ab/packages/contracts/src.ts/index.ts#L78

![Uploading event.png…]()

Reply:
As mentioned offline, I sort of think it might be cleaner and easier to just use the unique fields of these types to distinguish rather than the event name, which relies on an assertion.

Reply:
Resolved: https://github.com/across-protocol/sdk-v2/pull/501/commits/6953b372e9d237b4a64b67b46b08a828711f4c54